File: api/src/main/java/net/blueshell/api/dto/request/PasswordResetRequest.java
package net.blueshell.api.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.dto.BaseDTO;
import net.blueshell.api.validation.request.ValidPasswordResetRequest;
import net.blueshell.api.validation.user.ExistingUsername;
import net.blueshell.api.validation.request.ValidMemberActivationRequest;

@EqualsAndHashCode(callSuper = true)
@Data
@ValidPasswordResetRequest
public class PasswordResetRequest extends BaseDTO {

    @NotBlank
    private String token;

    @NotBlank
    @ExistingUsername
    private String username;

    @JsonProperty
    @Size(min = 8, max = 100, message = "Password must be at least 8 characters")
    @Pattern(
            regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]+$",
            message = "Password must contain at least one lowercase letter, one uppercase letter, one number, and one special character (@$!%*?&)"
    )
    private String password;
}

File: api/src/main/kotlin/net/blueshell/api/dto/recovery/PasswordResetRequest.kt
package net.blueshell.api.dto.recovery

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import net.blueshell.api.base.BaseDTO
@Schema(name = "PasswordResetRequest")
class PasswordResetRequest : BaseDTO() {
    @NotBlank
    val token: @NotBlank String? = null

    @NotBlank
    @Size(min = 8, max = 100, message = "Password must be at least 8 characters")
    @Pattern(
        regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]+$",
        message = "Password must contain at least one lowercase letter, one uppercase letter, one number, and one special character (@$!%*?&)"
    )
    val password: @NotBlank @Size(
        min = 8,
        max = 100,
        message = "Password must be at least 8 characters"
    ) @Pattern(
        regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]+$",
        message = "Password must contain at least one lowercase letter, one uppercase letter, one number, and one special character (@$!%*?&)"
    ) String? = null
}
